src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -774,6 +773,7 @@ class Peer(
                                         nodeParams.liquidityPolicy.value.maybeReject(request.walletInputs.balance.toMilliSatoshi(), totalFee, LiquidityEvents.Source.OnChainWallet, logger)?.let { rejected ->
                                             logger.info { "rejecting open_channel2: reason=${rejected.reason}" }
                                             nodeParams._nodeEvents.emit(rejected)
+                                            input.send(WalletCommand.UnlockWalletInputs(request.walletInputs.map { it.outPoint }.toSet()))
```

Comment:
I'm currently only unlocking utxos when funding attempts are rejected by our liquidity policy, but we could also unlock them if the funding fails later in the process, by having the corresponding channel state add a new action to unlock the inputs it was using. Do you think it's worth doing in this PR?